include/mbgl/ios/MGLShape.h

```diff
@@ -0,0 +1,14 @@
+#import <Foundation/Foundation.h>
+
+#import "MGLAnnotation.h"
+
+/** The `MGLShape` class is an abstract class that defines the basic properties for all shape-based annotation objects. This class must be subclassed and cannot be used as is. Subclasses are responsible for defining the geometry of the shape and providing an appropriate value for the coordinate property inherited from the `MGLAnnotation` protocol. */
+@interface MGLShape : NSObject <MGLAnnotation>
+
+/** The title of the shape annotation. The default value of this property is `nil`. */
+@property (nonatomic, copy) NSString *title;
```

Comment:
This property should be `nonnull` when we ultimately merge in #1578. It looks like the `MKAnnotation` protocol has been audited for nullability but the concrete implementations have not. So [`-[MKAnnotation title]`](https://developer.apple.com/library/ios/documentation/MapKit/Reference/MKAnnotation_Protocol/#//apple_ref/occ/intfp/MKAnnotation/title) comes with this note, which implies `nonnull`:

> Although this property is optional, if you support the selection of annotations in your map view, you are expected to provide this property. This string is displayed in the callout for the associated annotation view.

while corresponding properties in `MKShape` and other subclasses lack this note.